File: modules/core/src/main/java/com/eventhorizon/pulsar/core/packet/codec/PacketCodec.java
package com.eventhorizon.pulsar.core.packet.codec;

import com.eventhorizon.pulsar.core.packet.buffer.PacketByteBufferInput;
import com.eventhorizon.pulsar.core.packet.buffer.PacketByteBufferOutput;

public interface PacketCodec {
    void write(PacketByteBufferOutput buffer);
    void read(PacketByteBufferInput buffer);
}

File: modules/core/src/main/kotlin/com/eventhorizon/pulsar/core/packet/codec/PacketCodec.kt
package com.eventhorizon.pulsar.core.packet.codec

import com.eventhorizon.pulsar.core.packet.buffer.PacketByteBufferInput
import com.eventhorizon.pulsar.core.packet.buffer.PacketByteBufferOutput

interface PacketCodec {
    fun write(buffer: PacketByteBufferOutput)
    fun read(buffer: PacketByteBufferInput)
}